### PR TITLE
Add fallback icons

### LINKED ISSUE #199: Bundle Qt Gtk theme in AILL AppImage
Bundle Qt Gtk theme in AILL AppImage so that it looks native e.g., on Xfce and Gnome.

https://github.com/TheAssassin/AppImageLauncher/edit/master/travis/build-lite.sh should build

```
plugins/styles/libqgtk2style.so
plugins/platformthemes/libqgtk2.so
```

e.g., like this:

```
apt-get update
apt-get -y install libgtk2.0-dev
git clone http://code.qt.io/qt/qtstyleplugins.git
cd qtstyleplugins
qmake # be sure to use the one from the Qt installation you are using for the application
make -j$(nproc)
sudo make install 
cd -
```

Then,

```
plugins/styles/libqgtk2style.so
plugins/platformthemes/libqgtk2.so
```
need to be deployed into the AppImage.

### PR DESCRIPTION
**Pull request by azubieta**
Attempts to fix #199 but the qt-gtk-widgetsstyle is broken and ignores the icon theme preferences set by Qt. Nevertheless this pr do ensures that the icons are shown when the icons are missing on the target system and the default qt widgets style is used.

- Created a fallback icon theme (it gets ignored on the gtk style)

### CODE REVIEW

**Comment by TheAssassin** [comment]
Oh come on, seriously? ***So many changes*** just for a minor *look* issue? @probonopd seriously, this is just pure maintenance overhead...

**Comment by TheAssassin** [comment]
Where are these icons coming from? There's no license notice, so whose IP are those?

File: src/ui/settings_main.cpp [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
+#include <QIcon>
+#include <QDebug>
 #include <QApplication>
 
 // local
 #include "settings_dialog.h"
 
 int main(int argc, char** argv) {
+    // QIcon::setFallbackThemeName doesn't have effect on gtk styles 
+    QIcon::setThemeName("fallback");
```
Comment (TheAssassin) [inline src/ui/settings_main.cpp]: Doesn't this mean like, "use it anyway", and overwrite the theme for *all* platforms? This isn't sufficiently commented...

File: src/ui/CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 if(NOT BUILD_LITE)
     # main AppImageLauncher application
-    add_executable(AppImageLauncher main.cpp resources.qrc first-run.cpp first-run.h first-run.ui)
+    add_executable(AppImageLauncher main.cpp ../../resources/resources.qrc first-run.cpp first-run.h first-run.ui)
```
Comment (TheAssassin) [inline src/ui/CMakeLists.txt]: Please make the path a variable if it's no longer trivial.

File: src/ui/settings_dialog.ui [hunk 3]
```diff
@@ -138,7 +141,9 @@
    </item>
   </layout>
  </widget>
- <resources/>
+ <resources>
+  <include location="../../resources/resources.qrc"/>
```
Comment (TheAssassin) [inline src/ui/settings_dialog.ui]: Are you sure this is necessary? Or is this more like, for Qt Designer?

Comment (azubieta) [reply src/ui/settings_dialog.ui]: It's for Qt Designer.

File: src/ui/settings_main.cpp (hunk 1 above)
Comment (azubieta) [reply src/ui/settings_main.cpp]: Comments were added.

**Comment by TheAssassin** [comment]
Thank you for your work, but this leads to more and more code changes and workarounds. Let's just make a clean cut now.

File: src/ui/CMakeLists.txt (hunk 2 above)
Comment (azubieta) [reply src/ui/CMakeLists.txt]: done

**Comment by azubieta** [comment]
Wait, don't close this PR solve a real issue. Even in qt based system if the `gtk-ok`, `gtk-cancel` and `folder` icons are missing they will not be shown.

**Comment by TheAssassin** [comment]
Well, in that case we can keep the rest, too, I guess...

I'll let you work on this then. Let's see what the outcome will be.

**Comment by azubieta** [comment]
> Where are these icons coming from? There's no license notice, so whose IP are those?

Breeze icons where used. This icon theme package is licensed under LGPL (https://github.com/KDE/breeze-icons/blob/master/COPYING.LIB)
Releasing AppImageLauncher using these icons will force use to release the whole product using LGPL also.

We have to find a different set of icons.

**Comment by TheAssassin** [comment]
Then things look even less "native" than just using the regular Qt theme, whatever that is is up to the system.

**Comment by TheAssassin** [comment]
On my Xfce system, things are just perfectly fine as they are. I consider this native enough. As long as the user knows where to click... where's the problem?

**Comment by probonopd** [comment]
> On my Xfce system, things are just perfectly fine as they are. 

In case the screenshot you posted is from your Xfce system let me assure that it does not look Xfce native at all - colors are not native, fonts are not native, buttons are not native, icons are not native. In fact, there is nothing native about it.

**Comment by TheAssassin** [comment]
Not entirely true. The font *is* the same as on the rest of the system. And: the whole thing looks like many other Qt apps, so you could call it "native".

If someone wants nativeness, they shall install the packages. If someone wants the AppImage, well, this is as good as we can get.

After all, functionality is what counts, not something as unimportant as "wrong window color"...

**Comment by probonopd** [comment]
> the whole thing looks like many other Qt apps, so you could call it "native".

Well, my definition of "native" is that it looks like the applications that come with the base system, not that it looks like other applications that did not come with the base system either. (By your definition, you could call all Java applications look "native" because they, well, "look like Java apps"...)

**Pull request closed without merging**